extensions/mapreduce/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -121,20 +126,31 @@
    @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
    private MapReducer<KOut, VOut, R> mapReducer;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Clustered clustered;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private CacheInformation cacheInformation;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private BasicOperations basicOperations;
 
    @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Iterable iterable;
 
    private boolean supportsResultCacheName = false;
 
+   @Init
+   public void validate() {
```

Comment:
@alanfx I agree with @rvansa here. As long as we're moving MR to a separate module, we shouldn't impose storing the results in a cache as we want to make the extension as generic as possible.